### ISSUE TITLE
Maximum call size exceeded error 

### ISSUE THREAD

**Issue by avishekgurung** [Meteor-Community-Packages/meteor-autoform#1023]
This is a very simple application. I do not know why I am getting "Uncaught RangeError: Maximum call stack size exceeded". 

In folder called 'both', I have a JS file with schema defined as follows

```
Schema1 = new SimpleSchema({
    name:{
        type:String,
        label:'Name'
    }
});
```

In client folder,
Now I have an HTML template as shown:

```
<template name="test">
    {{#autoForm schema="Schema1" id = "schema1" type="method-update" meteormethod="processInServer" doc=formDoc}}
        {{> afQuickField name='name'}}
        <button type="insert">Update</button>
    {{/autoForm}}
</template>
```

The helpers of the above template is as follows:

```
Template.test.helpers({
    formDoc:function(){
        return 'avishek';
    }
})

var hook = {
    before:{
        method:function(doc){
            console.log(doc);
            return doc;
        }
    }
}

AutoForm.addHooks(['schema1'],hook);
```

In server folder,

```
Meteor.methods({
    'processInServer': function () {
        console.log('called server');
        return 'success';
    }
})
``

What could go wrong in this application. I have nothing apart from this.
```

**Comment by jmlv** [comment]
use doc=this.doc

![2015-06-09 14-02-46 1 node 80x24](https://cloud.githubusercontent.com/assets/7571432/8056412/6efd890a-0eb0-11e5-8bd2-521798c0e5ef.png)

**Comment by jononomo** [comment]
I am running into the same error - "Maximum call stack size exceeded".  I don't understand why it is happening and I don't understand the comment from @jmlv either.

**Comment by samybob1** [comment]
I have an insert form and the error occurs when the form is displayed. I have 14 fields in my schema. The error does not occur anymore when I remove a few fields.

Here is the stack trace:

> Exception from Tracker afterFlush function:
> RangeError: Maximum call stack size exceeded
>     at Object (native)
>     at Function._.isObject (http://localhost:3000/packages/underscore.js:1063:20)
>     at http://localhost:3000/packages/[aldeed_autoform.js:322:48](https://github.com/aldeed/meteor-autoform/blob/16f100ce96a13823b293414f5c615ea67693a39e/utility.js#L252)
>     at Function._.each._.forEach (http://localhost:3000/packages/underscore.js:157:22)
>     at compactArrays (http://localhost:3000/packages/[aldeed_autoform.js:316:9](https://github.com/aldeed/meteor-autoform/blob/16f100ce96a13823b293414f5c615ea67693a39e/utility.js#L246))
>     at http://localhost:3000/packages/[aldeed_autoform.js:324:11](https://github.com/aldeed/meteor-autoform/blob/16f100ce96a13823b293414f5c615ea67693a39e/utility.js#L254)
>     at Function._.each._.forEach (http://localhost:3000/packages/underscore.js:157:22)
>     at compactArrays (http://localhost:3000/packages/[aldeed_autoform.js:316:9](https://github.com/aldeed/meteor-autoform/blob/16f100ce96a13823b293414f5c615ea67693a39e/utility.js#L246))
>     at http://localhost:3000/packages/[aldeed_autoform.js:324:11](https://github.com/aldeed/meteor-autoform/blob/16f100ce96a13823b293414f5c615ea67693a39e/utility.js#L254)
>     at Function._.each._.forEach (http://localhost:3000/packages/underscore.js:157:22)

**Comment by oskareke** [comment]
This happens to me as well after updating to Meteor 1.3.

**Comment by Amritpd** [comment]
using this.doc seemed to have fixed the issue. If you have any routes in place then having "doc" by itself means something else. this.doc tells Meteor to specifically use the helper function attached to that template called "doc"

**Comment by thsowers** [comment]
Having the same issue... @Amritpd You're saying that passing this.doc to the helper function fixes this?

**Comment by thsowers** [comment]
Fixed my problem...rookie mistake: turned out I was returning an doc ID to the helper, not an actual document

**Comment by Amritpd** [comment]
@thsowers Good to know you fixed your problem! For some reason, simply specifying type=update wasn't enough for AutoForm to pre-populate the form with the current object's values. Furthermore, after pressing the "Save Changes" button, the object was not updated in the collection. I had to come up with an alternative way to fill the update form with the object's data and ensure it gets updated properly. My solution ended up being a bit more complicated but it works:

       {{#with currentRow}}
	{{#autoForm id="editPaymentForm" type="method-update" meteormethod="updateTable" collection="paymentTable" doc=currentRow}}`
         <fieldset>
		{{> afQuickField name="dateReceived" value=dateReceived}}
		{{> afQuickField name="paymentAmount" label="Payment Amount" value=payment}}
		{{> afQuickField name="taxes" value=taxes}}
		{{> afQuickField name="fees" value=fees}}
		<button type="submit" class="btn btn-primary">Save Changes</button>
		<button class="cancel">Cancel</button>
		 </fieldset>
		{{/autoForm}}
		{{/with}}

What's happening in the code above is that I created a helper function called currentRow that returns the object I want to edit. This object included its _id, and various other attributes like taxes or fees. Notice that I don't need to use doc=this.currentRow anymore. 

I registered a Handlebars helper called "with" that allows me to directly use the values of the object. Its implementation can be found [here](http://handlebarsjs.com/block_helpers.html). By using type=method-update, I could manually update the object.
